struct.c

```diff
@@ -899,21 +904,25 @@ rb_struct_each(VALUE s)
 
 /*
  *  call-seq:
- *     struct.each_pair {|sym, obj| block }     -> struct
- *     struct.each_pair                         -> enumerator
+ *    each_pair {|name, value| ... } -> self
```

Comment:
This is subtly incorrect.  `Struct#each_pair` actually yields a single array, not two separate arguments:

```
$ ruby -e 'Struct.new(:a).new.each_pair(&->(a, b){p a, b})'
-e:1:in `block in <main>': wrong number of arguments (given 1, expected 2) (ArgumentError)
```

It works with a normal block, but not with a lambda, because a normal block will autosplat.

In terms of documenting this, you can use:

```
each_pair {|(name, value)| ... } -> self
```

This will deconstruct the single argument into two arguments, and work with both normal blocks and lambdas.

Reply:
Done.